### ISSUE TITLE
Many Pylint Errors - some of which are bugs

### ISSUE THREAD

**Issue by galewis2** [Esri/arcgis-python-api#1387]
**Describe the bug**
In chasing down some bugs in the python api, I decided to run Pylint against it to see if there was anything that would explain my issue. And even filtering out a bunch of the stylistic and non-issues, I came back with a large report.

**To Reproduce**
(Using latest Notebook Docker image)
pylint --disable=I,R,C,W,E0401,F,E0611 arcgis

[report.txt](https://github.com/Esri/arcgis-python-api/files/10015250/report.txt)

For example, that finds:

![image](https://user-images.githubusercontent.com/62433564/201997658-124cf004-5548-4cb0-b588-ed18df016692.png)

**Expected behavior**
Snag some of the legitimate issues that Pylint highlights

**Platform (please complete the following information):**
- ArcGIS Python 2.0.1 in esridocker/arcgis-api-python-notebook:latest

**Comment by nanaeaubry** [comment]
Thanks for reporting this we will take a look!

**Comment by nanaeaubry** [comment]
@galewis2 Your report has helped us identify a few small bugs and typos so thank you for submitting it. However, many of the issues found by pylint are 'false issues' since they don't always take into account the full context. For example, there were many issues brought up about some attribute not being a member of a variable but pylint doesn't realize that the variable is an instance of a class and thus the member does exist. This is just to let you know why it seems there are so many rather than just a handful that we were able to fix.

I think these errors can be avoided by implementing more and more strict typing, which we are integrating more of at each release. 


@priyankatuteja I did not address any of the issues brought up for the learn module in the report above. Feel free to take a look if you would like but I have looked at all other modules just FYI.
